pandas/core/dtypes/missing.py

```diff
@@ -233,9 +233,10 @@ def _isna_ndarraylike_old(obj):
     values = getattr(obj, "_values", obj)
     dtype = values.dtype
 
-    if is_string_dtype(dtype):
+    if is_extension_array_dtype(dtype):
+        result = values.isna() | (values == -np.inf) | (values == np.inf)
```

Comment:
Is this always treating inf as NA?

Reply:
Yes, I think / hope that's the intention when this gets called (e.g., at line 244 we have `result = ~np.isfinite(values)`)

Reply:
can you make the similar code paths in isna_ndarraylike an here

Reply:
The functions are almost the same now, unless there are objections (can revert if so) I'm going to try parametrizing _isnda_ndarraylike over old / new and just deleting this function